js/data/line_bucket.js

```diff
@@ -63,7 +68,7 @@ LineBucket.prototype.shaders = {
                 // The z component's first bit, as well as the sign bit is reserved for the direction,
                 // so we need to shift the linesofar.
                 '((dir < 0) ? -1 : 1) * ((dir ? 1 : 0) | ((linesofar << 1) & 0x7F))',
-                '(linesofar >> 6) & 0x7F'
+                '((linesofar >> 6) & 0xFF) - 128'
```

Comment:
Is this complex bit packing really necessary? If so, can you add a comment to the source about why it is necessary and how it works.